src/plugins/discover/public/application/discover_router.tsx

```diff
@@ -18,7 +18,15 @@ import { NotFoundRoute } from './not_found';
 import { DiscoverServices } from '../build_services';
 import { ViewAlertRoute } from './view_alert';
 
-export const discoverRouter = (services: DiscoverServices, history: History) => (
+export interface DiscoverRouterProps {
+  isDev: boolean;
+}
+
+export const discoverRouter = (
+  services: DiscoverServices,
+  history: History,
+  { isDev }: DiscoverRouterProps
```

Comment:
Just an opinion, introducing 3rd argument as a boolean should be fine, if there were more, it might make sense to convert to use DiscoverRouterProps with `services`, `history`, `isDev` and `crazyNewProperty` 
```suggestion
  isDev: boolean
```